test/component/Legend.spec.tsx

```diff
@@ -2930,6 +2992,49 @@ describe('<Legend />', () => {
     });
   });
 
+  describe('legend portal', () => {
+    it('should render outside of SVG, as a direct child of recharts-wrapper by default', () => {
+      const { container } = render(
+        <ScatterChart width={500} height={500} data={numericalData}>
+          <Legend />
+          <Scatter dataKey="percent" />
+        </ScatterChart>,
+      );
+
+      expect(container.querySelectorAll('.recharts-wrapper svg .recharts-legend-wrapper')).toHaveLength(0);
+      expect(container.querySelector('.recharts-wrapper > .recharts-legend-wrapper')).toBeVisible();
+    });
+
+    it('should render in a custom portal if "portal" prop is set', () => {
```

Comment:
Do you think it's valuable to test what happens when `portal` prop is undefined? Or some other invalid value?

Reply:
Yeah sure, I can add something. Although if it's undefined it uses the one from the chart

Reply:
done